### PR TITLE
[FIX] allow direct initialisation with two_dimensional_matrix

### PR DESCRIPTION
**Pull request by marehr**

### CODE REVIEW

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ class two_dimensional_matrix
      */
     template <std::ranges::forward_range entries_t>
     //!\cond
-        requires (!std::same_as<entries_t, storage_type>) && (std::convertible_to<value_type_t<entries_t>, value_type>)
+        requires (std::convertible_to<value_type_t<entries_t>, value_type>)
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]:
```suggestion
        requires (std::convertible_to<std::ranges::value_type_t<entries_t>, value_type>)
```

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp [hunk 2]
```diff
@@ -117,7 +117,7 @@ class two_dimensional_matrix
      */
     template <std::ranges::forward_range entries_t>
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]: documentation of this template argument is missing
also `\param` is missing the `[in]` specifier

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp [hunk 3]
```diff
@@ -117,7 +117,7 @@ class two_dimensional_matrix
...
     //!\cond
-        requires (!std::same_as<entries_t, storage_type>) && (std::convertible_to<value_type_t<entries_t>, value_type>)
+        requires (std::convertible_to<value_type_t<entries_t>, value_type>)
     //!\endcond
     two_dimensional_matrix(number_rows const row_dim, number_cols const col_dim, entries_t entries) :
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]: why is entries not const?

**Comment by marehr** [comment]
> Do I understand correctly that now
> 
> ```
> matrix1{number_rows{0}, number_cols{0}, {}}
> ```
> 
> is now possible because `{}` ist implicitely cast to `storage_type` while before the type could not be deduced?

Yes

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp (hunk 2 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]: :sob:

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp (hunk 3 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]: All it's elements get moved and modified. I guess this should be non-const.

File: include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]:
```
/seqan3/include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp:121:52: error: ‘value_type_t’ is not a member of ‘std::ranges’
  121 |         requires (std::convertible_to<std::ranges::value_type_t<entries_t>, value_type>)
      |                                                    ^~~~~~~~~~~~
```

Need further instructions

And ranges says:

```c++
    template<typename T>
    using value_type_t RANGES_DEPRECATED(
        "ranges::value_type_t is deprecated. Please use "
        "ranges::iter_value_t instead.") = iter_value_t<T>;
```

Comment (smehringer) [reply include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp]: Hm I am confused :smile: I thought I had this specific review comment before in one of my PRs but I see now that we do not have my suggestion in the code base... sorry. I'll merge it as is.

**Pull request merged**